File: frontend/src/app/core/models/participant.dto.ts
export interface ParticipantDto {
  name: string;
  tag: string;
  puuid: string;

  champion: string;
  championIcon: string;
  championSplashArt: string;

  kills: number;
  deaths: number;
  assists: number;
  win: boolean;

  goldEarned: number;
  goldSpent: number;

  damageDealtToChampions: number;
  totalDamageDealt: number;
  damageTaken: number;

  visionScore: number;
  wardsPlaced: number;
  wardsKilled: number;

  cs: number;
  kdaRatio: number;
  level: number;

  turretKills: number;
  inhibitorKills: number;
}

File: backend/src/main/java/com/lol/backend/dto/ParticipantDto.java
package com.lol.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ParticipantDto {
    private String name;
    private String tag;
    private String puuid;

    private String champion;
    private String championIcon;
    private String championSplashArt;

    private int kills;
    private int deaths;
    private int assists;
    private boolean win;

    private int goldEarned;
    private int goldSpent;

    private int damageDealtToChampions;
    private int totalDamageDealt;
    private int damageTaken;

    private int visionScore;
    private int wardsPlaced;
    private int wardsKilled;

    private int cs;
    private double kdaRatio;
    private int level;

    private int turretKills;
    private int inhibitorKills;
}